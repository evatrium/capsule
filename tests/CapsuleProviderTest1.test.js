import React from 'react';
import renderer from 'react-test-renderer';

import {CapsuleProvider, events} from "./_test_utils/cap1/exports";

import {Till} from "./_test_utils";

const theme = {
    color: {
        primary: 'red',
    },
    mixins: {
        centerize: {
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center'
        },
        all100: {
            height: '100%',
            width: '100%'
        }
    }
};

const till = Till(events);

describe('CapsuleProvider', () => {


    it(`
asynchronously loads the app without crashing.
verify that:
    theme is passed
    classes are created
    toProps passes data to props
    toLogic passes logic to logic
    routing is available from collective in logic
    `, async () => {

        const App = CapsuleProvider({
            loadApp: () => import('./_test_utils/cap1/LazyLoaded'),
            theme,
            toProps: {propsTest: 'props from provider'},
            toLogic: {toLogicTest: 'logic from provider'},
        })();

        const tree = renderer.create(<App/>);

        await till('mounted').then((data) => {
            expect(data.cn).toBeDefined();
            expect(data.theme.color.primary).toBe('red');
            expect(data.propsTest).toBe('props from provider');
            expect(data.toLogicTest).toBe('logic from provider');
            expect(data.routing).toBeInstanceOf(Object)
        });

        expect(tree).toMatchSnapshot();

    });


});
