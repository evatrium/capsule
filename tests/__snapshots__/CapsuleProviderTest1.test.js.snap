// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CapsuleProvider 
asynchronously loads the app without crashing.
verify that:
    theme is passed
    classes are created
    toProps passes data to props
    toLogic passes logic to logic
    routing is available from collective in logic
     1`] = `
Array [
  <div
    className="x0 x1 x2 x3 x4"
  >
    <button
      className="x5"
    >
      loggin
    </button>
    {
    "theme": {
        "color": {
            "primary": "red"
        },
        "mixins": {
            "centerize": {
                "display": "flex",
                "alignItems": "center",
                "justifyContent": "center"
            },
            "all100": {
                "height": "100%",
                "width": "100%"
            }
        }
    },
    "propsTest": "props from provider",
    "classes": {
        "app": "x0 x1",
        "someOtherClass": "x2 x3 x4",
        "loginButton": "x5"
    }
}
  </div>,
  " ",
]
`;
